src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -487,55 +470,41 @@ async fn extract_output_file() {
     .store_file_bytes(TestData::roland().bytes(), false)
     .await
     .expect("Error saving file bytes");
-  store
-    .record_directory(&TestDirectory::containing_roland().directory(), true)
-    .await
-    .expect("Error saving directory");
-  let directory_digest = store
-    .record_directory(&TestDirectory::nested().directory(), true)
-    .await
-    .expect("Error saving directory");
+  let input_tree = TestTree::nested();
 
-  let file_node = crate::remote_cache::CommandRunner::extract_output_file(
-    directory_digest,
-    RelativePath::new("cats/roland.ext").unwrap(),
-    &store,
+  let output_file = crate::remote_cache::CommandRunner::extract_output_file(
+    &input_tree.digest_trie(),
+    "cats/roland.ext",
   )
-  .await
   .unwrap()
   .unwrap();
 
-  // Note that the `FileNode` only stores the file name, but we will end up storing the full path
-  // in the final ActionResult.
-  assert_eq!(file_node.name, "roland.ext");
-  let file_digest: Digest = file_node.digest.unwrap().try_into().unwrap();
+  assert_eq!(output_file.path, "cats/roland.ext");
+  let file_digest: Digest = output_file.digest.unwrap().try_into().unwrap();
   assert_eq!(file_digest, TestData::roland().digest());
 
   // Extract non-existent files to make sure that Ok(None) is returned.
   assert!(crate::remote_cache::CommandRunner::extract_output_file(
-    directory_digest,
-    RelativePath::new("animals.ext").unwrap(),
-    &store,
+    &input_tree.digest_trie(),
+    "animals.ext",
   )
-  .await
   .unwrap()
   .is_none());
   assert!(crate::remote_cache::CommandRunner::extract_output_file(
-    directory_digest,
-    RelativePath::new("cats").unwrap(),
-    &store,
+    &input_tree.digest_trie(),
+    "cats/xyzzy",
   )
-  .await
-  .unwrap()
-  .is_none());
-  assert!(crate::remote_cache::CommandRunner::extract_output_file(
-    directory_digest,
-    RelativePath::new("cats/xyzzy").unwrap(),
-    &store,
-  )
-  .await
   .unwrap()
   .is_none());
+
+  // Error if a path has been declared as a file but isn't.
```

Comment:
@tdyas : This is a behavior change, but a reasonable one I think?

Reply:
And follows the spec. If a path is declared via `output_files` and is a directory, then we should error. (And vice versa for a directory declared via `output_directories`, that is actually a file.)

Note that `output_files` and `output_directories` are deprecated in favor of `output_paths` which does not have need for this error behavior. But Pants doesn't use `output_paths`, so point is moot.